docs/source/how-to-upstream.mdx

```diff
@@ -24,6 +26,34 @@ Whenever you want to upload files to the Hub, you need to log in to your Hugging
 
    [`notebook_login`] will launch a widget in your notebook from which you can enter your Hugging Face credentials.
 
+## Upload a file
+
+Use the [`upload_file`] function to upload a file to a repository. Let the function know where the file you want to upload is, where you want to upload the file to in the repository, and the name of the repository you want to add the file to.
+
+```py
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+>>> api.upload_file(path_or_fileobj="/home/lysandre/dummy-test/README.md", 
+...                 path_in_repo="README.md", 
+...                 repo_id="lysandre/test-model",
+... )
+```
+
+## Upload a folder
+
+Use the [`upload_folder`] function to upload a local folder to a repository. Specify the path of the local folder to upload, where you want to upload the folder to in the repository, and the name of the repository you want to add the folder to.
+
+```py
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+>>> api.upload_folder(folder_path="/path/to/local/folder",
+...                   path_in_repo="my-dataset/train",
+...                   repo_id="username/test-dataset"
+...                   repo_type="datasets",
```

Comment:
In the code snippet above, `repo_type` isn't used, but here it is.

I'd say either use it for both (and explain what it does), or for neither.